src/rgw/rgw_process.cc

```diff
@@ -401,16 +401,18 @@ int process_request(rgw::sal::Store* const store,
   }
 
 done:
-  if (op && s->trace) {
-    s->trace->SetAttribute(tracing::rgw::RETURN, op->get_ret());
-    if (s->user) {
-      s->trace->SetAttribute(tracing::rgw::USER_ID, s->user->get_id().id);
-    }
-    if (s->bucket) {
-      s->trace->SetAttribute(tracing::rgw::BUCKET_NAME, s->bucket->get_name());
-    }
-    if (s->object) {
-      s->trace->SetAttribute(tracing::rgw::OBJECT_NAME, s->object->get_name());
+  if (op) {
+    if (s->trace) {
+      s->trace->SetAttribute(tracing::rgw::RETURN, op->get_ret());
+      if (rgw::sal::User::empty(s->user)) {
+        s->trace->SetAttribute(tracing::rgw::USER_ID, s->user->get_id().id);
+      }
+      if (rgw::sal::Bucket::empty(s->bucket)) {
+        s->trace->SetAttribute(tracing::rgw::BUCKET_NAME, s->bucket->get_name());
+      }
+      if (rgw::sal::Object::empty(s->object)) {
```

Comment:
oh, oops. these should all be checking for `!empty()`. you're probably dereferencing nullptrs here

Reply:
Great! We're just collecting the [mgr logs](https://jenkins.ceph.com/job/ceph-api/40655/artifact/build/out/) in the Jenkins runs, but if you think it's useful, please feel free to extend this to other logfiles too:

https://github.com/ceph/ceph-build/blob/0aa19992ffb5c7b38658d917dd154e73413f0d97/ceph-pr-api/config/definitions/ceph-pr-api.yml#L88